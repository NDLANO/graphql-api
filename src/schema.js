/**
 * Copyright (c) 2018-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

const { makeExecutableSchema } = require('graphql-tools');
const resolvers = require('./resolvers');

const typeDefs = `

type ResourceType {
  id: String,
  name: String,
}

type Resource {
  id: String!,
  name: String!,
  contentUri: String,
  path: String!
  article: Article,
  resourceTypes: [ResourceType]
}

type Article {
  title: String!,
  content: String!,
  introduction: String!,
  metaDescription: String!,
  resourceTypes: [ResourceType]
}

type ArticleSubset {
  title: String!,
  introduction: String!,
  metaDescription: String!,
}

type Description {
  value: String!,
}

type Topic {
  id: String!,
  contentUri: String,
  name: String!,
  path: String!
  article: Article,
  meta: ArticleSubset
}

type Subject {
  id: String!,
  contentUri: String,
  name: String!,
  path: String!,
  topics: [Topic]
}


type Query {
  resource(id: String!): Resource
  article(id: String!): Article
  subject(id: String!): Subject
  subjects: [Subject]
  topic(id: String!): Topic
  topics: [Topic]
}

`;

const schema = makeExecutableSchema({ typeDefs, resolvers });

module.exports = schema;
