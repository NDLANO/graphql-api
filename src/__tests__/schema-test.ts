/**
 * Copyright (c) 2018-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import { typeDefs } from '../schema';
import { makeExecutableSchema, addMockFunctionsToSchema } from 'graphql-tools';
import { graphql } from 'graphql';

test('can run query on schema', async () => {
  const schema = makeExecutableSchema({
    typeDefs,
    resolverValidationOptions: { requireResolversForResolveType: false },
  });
  addMockFunctionsToSchema({
    schema,
  });
  const query = `
    query resource {
      resource(id: "6") { id, name }
    }
  `;

  const result = await graphql(schema, query);

  expect(result).toMatchSnapshot();
});
