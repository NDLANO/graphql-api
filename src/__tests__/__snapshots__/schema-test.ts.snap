// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can run query on schema 1`] = `
Object {
  "data": Object {
    "resource": Object {
      "id": "Hello World",
      "name": "Hello World",
    },
  },
}
`;
